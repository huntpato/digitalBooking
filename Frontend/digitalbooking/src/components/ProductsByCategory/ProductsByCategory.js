import React from "react";
import { ReCard } from "../ReCard/ReCard";
import useFetch from "use-http";
import styles from "./ProductsByCategory.module.css";

export const ProductsByCategory = (props) => {
  const options = {headers:{
    "X-Requested-With": "XMLHttpRequest"
  }};
  const {category, name} = props; 
  const url = "https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/products";
  const { loading, data = [] } = useFetch(url, options, []);
  const { recommendation, recommendationCards, recommendationtitle, recommendationContainer } = styles;

  const filteredData = data.filter(location => location.productCategory.id == category)

  return (
    <div className={recommendationContainer}>
      <section className={recommendation}>
        <h4 className={recommendationtitle}>Recomendaciones de {name}</h4>
        <div className={recommendationCards}>
          {loading && <p>Cargando Locaciones...</p>}
          {filteredData.slice(0,5).map((product, index) => (
            <ReCard
              key={index}
              type={product.productCategory.name}
              title={product.productName}
              img={product.photoGallery[0]}
              alt={product.productDescription}
              text={product.productDescription}
              loc={product.location}
              rate={product.review}
              id={product.id}
              services= {product.services}
              latlong={product.coordinates}
            />
          ))}
        </div>
      </section>
    </div>
  );
};
