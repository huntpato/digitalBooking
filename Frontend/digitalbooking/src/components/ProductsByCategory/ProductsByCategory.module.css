.recommendationContainer{
  background-color: var(--background-color2);
  display: flex;
  justify-content: center;
  width:100%;

}

.recommendation {
  width: 100%;
  background-color: var(--background-color2);
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: space-around;
  padding: 30px 0px 20px 0px;
  width: 90vw;
}

.recommendationtitle {
  margin-bottom: 10px;
}

.recommendationCards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
}

@media (max-width: 768px) {

  .recommendationCards {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
}

@media (max-width: 480px) {
}
