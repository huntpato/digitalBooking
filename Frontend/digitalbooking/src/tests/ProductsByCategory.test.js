import React from 'react';
import { shallow } from 'enzyme';
import { ProductsByCategory } from '../components/ProductsByCategory/ProductsByCategory'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente  ProductsByCategory />', ()=> {
   
    const wrapper= shallow( <ProductsByCategory />);

    test('Comprobar que  ProductsByCategory /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const section = wrapper.find('section');
        expect(section.hasClass('recommendation')).toBe(true);
    })

})
