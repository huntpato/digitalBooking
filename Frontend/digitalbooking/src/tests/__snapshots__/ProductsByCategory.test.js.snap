// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente  ProductsByCategory /> Comprobar que  ProductsByCategory /> renderiza correctamente. 1`] = `
<div
  className="recommendationContainer"
>
  <section
    className="recommendation"
  >
    <h4
      className="recommendationtitle"
    >
      Recomendaciones de 
    </h4>
    <div
      className="recommendationCards"
    >
      <p>
        Cargando Locaciones...
      </p>
    </div>
  </section>
</div>
`;
